File: src/main/java/com/github/joelws/release/tracker/dao/ArtifactDao.java
package com.github.joelws.release.tracker.dao;

import com.github.joelws.release.tracker.common.GenericDao;
import com.github.joelws.release.tracker.model.Artifact;

import java.util.List;


public interface ArtifactDao extends GenericDao<Artifact, String>
{

        List<Artifact> findAllById(final String id);
}

File: src/main/kotlin/com/joelws/release/tracker/dao/artifact/ArtifactDao.kt
package com.joelws.release.tracker.dao.artifact

import com.joelws.release.tracker.dao.GenericDao
import com.joelws.release.tracker.entity.artifact.Artifact
import com.joelws.release.tracker.entity.artifact.ArtifactPK

interface ArtifactDao : GenericDao<Artifact, ArtifactPK> {
    fun list(): List<Artifact>
}
